sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -414,11 +414,11 @@ def run_pipeline(self, pipeline, options):
         logging.fatal(
             'Typical end users should not use this worker jar feature. '
             'It can only be used when fnapi is enabled.')
-
-      experiments = ["use_staged_dataflow_worker_jar"]
-      if debug_options.experiments is not None:
-        experiments = list(set(experiments + debug_options.experiments))
-      debug_options.experiments = experiments
+      else:
+        experiments = ["use_staged_dataflow_worker_jar"]
```

Comment:
Please use `add_experiment`: https://github.com/apache/beam/blob/98faabff70b58935b7cbc7efeacc8ec1a850479d/sdks/python/apache_beam/options/pipeline_options.py#L660

Reply:
Also please add a unit test.

Reply:
See: https://github.com/apache/beam/blob/2581565fe96f2a1b4be01993831898afefb3e4c7/sdks/python/apache_beam/runners/dataflow/dataflow_runner_test.py#L402

Reply:
Also, you need to rebase to get past https://github.com/apache/beam/pull/8225.

Reply:
Added unit tests also rebased. PTAL.